src/main/routes/receiver.ts

```diff
@@ -139,11 +139,12 @@ export default express.Router()
                                res: express.Response,
                                next: express.NextFunction): Promise<void> => {
       const cookies = new Cookies(req, res)
+      let user
 
       try {
         const authenticationToken = await getAuthToken(req)
         if (authenticationToken) {
-          const user = await IdamClient.retrieveUserFor(authenticationToken)
+          user = await IdamClient.retrieveUserFor(authenticationToken)
```

Comment:
do you plan to remove `AppPaths.linkDefendantReceiver.uri` in a separate PR?
as that was one of the goals of this

Reply:
I'd rather do it separately. 

What's more we will have to keep part of it (at least to redirect) as this url is stored in db:
![image](https://user-images.githubusercontent.com/5647337/35044974-35bbd498-fb8a-11e7-8eba-c8ea9a77d131.png)

in `continue_url` column, idam db.

Reply:
Also I think we can keep the `const` here, I can't see any reassignments in the remaining part of this call handler.

Reply:
@yarpo hmm, I think we could get away without it, or if we think we need it just add a redirect that to the new url making sure the query string parameters are preserved

Reply:
@rlewan we cannot :(
![image](https://user-images.githubusercontent.com/5647337/35047941-74667ea6-fb93-11e7-801f-ee92b740309c.png)

Reply:
@yarpo Why do you need the user extracted at that level?

Reply:
@timja-kainos yes, as I need it here:

```
const letterHolderId: string = getLetterHolderId(req, user)
          if (letterHolderId && user.isInRoles(`letter-${letterHolderId}`)) {
            const claim: Claim = await ClaimStoreClient.retrieveByLetterHolderId(letterHolderId)
            logger.debug(`Linking user ${user.id} claim ${claim.id}`)
            if (!claim.defendantId) {
              await ClaimStoreClient.linkDefendant(claim.id, user.id)
            }
          }
```